javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -133,24 +133,21 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     /**
      * Creates a TreeMap of the given list of key-value pairs.
      *
+     * @deprecated Should not be used any more because unsafe
+     *
      * @param pairs A list of key-value pairs
      * @param <K>   The key type
      * @param <V>   The value type
      * @return A new Map containing the given entries
      */
+    @Deprecated
     @SuppressWarnings("unchecked")
     public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Object... pairs) {
-        return of((Comparator<? super K> & Serializable) K::compareTo, pairs);
-    }
-
-    @SuppressWarnings("unchecked")
-    public static <K, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, Object... pairs) {
```

Comment:
No errors here... Which version do you use? I have `Java 1.8.0_101`.

The following code does work:

```java
public class Test {

    public static void main(String[] args) {

        // TreeMap((1, a), (2, b), (3, c), (4, d), (5, e), (6, f), (7, g), (8, h), (9, i), (10, j))
        final Map<Integer, String> map = TreeMap.of(Integer::compareTo,
                1, "a",
                2, "b",
                3, "c",
                4, "d",
                5, "e",
                6, "f",
                7, "g",
                8, "h",
                9, "i",
                10, "j"
                );

        // TreeMap((1, a), (2, b), (3, c), (4, d), (5, e), (6, f), (7, g), (8, h), (9, i), (10, j))
        System.out.println(map);

    }
}
```

I tend to keep the method and add a unit test in TreeMapTest. If the CI build is ok, we can keep it.